plugins/modules/identity/keycloak/keycloak_clientscope.py

```diff
@@ -0,0 +1,475 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_clientscope
+
+short_description: Allows administration of Keycloak client_scopes via Keycloak API
+
+description:
+    - This module allows you to add, remove or modify Keycloak client_scopes via the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+      be returned that way by this module. You may pass single values for attributes when calling the module,
+      and this will be translated into a list suitable for the API.
+
+    - When updating a group, where possible provide the group ID to the module. This removes a lookup
+      to the API to translate the name into the group ID.
+
+
+options:
+    state:
+        description:
+            - State of the client_scope.
+            - On C(present), the group will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the group will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    name:
+        type: str
+        description:
+            - Name of the client_scope.
+            - This parameter is required only when creating or updating the client_scope.
+
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this group resides.
+        default: 'master'
+
+    id:
+        type: str
+        description:
+            - The unique identifier for this client_scope.
+            - This parameter is not required for updating or deleting a client_scope but
+              providing it will reduce the number of API calls required.
+
+    description:
+        type: str
+        description:
+            - Description for this client_scope.
+            - This parameter is not required for updating or deleting a client_scope.
+
+    protocol:
+        description:
+            - Type of client (either C(openid-connect), C(saml) or C(wsfed)).
```

Comment:
No need to list the choices twice.
```suggestion
            - Type of client.
```